binr/r2r/r2r.c

```diff
@@ -853,20 +853,33 @@ static void fixup_tests(RPVector *results, const char *edited_file, ut64 start_l
 	}
 }
 
-static void replace_cmd_kv(const char *path, ut64 line_begin, ut64 line_end, const char *key, const char *value, RPVector *fixup_results) {
-	char *content = r_file_slurp (path, NULL);
-	if (!content) {
-		eprintf ("Failed to read file \"%s\"\n", path);
-		return;
-	}
+static char *replace_cmd_kv(const char *path, const char *content, ut64 line_begin, ut64 line_end, const char *key, const char *value, RPVector *fixup_results) {
 	char *kv = format_cmd_kv (key, value);
 	if (!kv) {
-		free (content);
-		return;
+		return NULL;
 	}
-	ut64 kv_lines = r_str_char_count (kv, '\n');
+	ut64 kv_lines = r_str_char_count (kv, '\n') + 1;
 	char *newc = replace_lines (content, line_begin, line_end, kv);
 	free (kv);
+	if (!newc) {
+		return NULL;
+	}
+	ut64 lines_before = line_end - line_begin;
+	st64 delta = (st64)kv_lines - lines_before;
+	if (line_end == line_begin) {
+		delta++;
+	}
+	fixup_tests (fixup_results, path, line_end, delta);
+	return newc;
+}
+
+static void replace_cmd_kv_file(const char *path, ut64 line_begin, ut64 line_end, const char *key, const char *value, RPVector *fixup_results) {
```

Comment:
i think line counts should be size_t instead of ut64

Reply:
The line counts are ut64 everywhere in r2r. The idea is that in theory line numbers could be more that 32bit because they are part of a file on the disk. Of course this will never happen as long as they are slurped into memory but imo it makes more sense this way.

Reply:
i know, but i think is more correct to go for size_t to count lines, thats why im suggesting you to do the change

Reply:
Replaced ut64 by size_t in the places where it is directly operating on memory as opposed to files.

Reply:
exactly